File: Base/Desktop/Allors/Client/Base/Angular/Result.ts
﻿namespace Allors {
    export class Result {

        objects: { [name: string]: ISessionObject; } = {};
        collections: { [name: string]: ISessionObject[]; } = {};
        values: { [name: string]: any; } = {};

        constructor(session: ISession, response: Data.Response) {
            _.map(response.namedObjects, (v, k) => {
                this.objects[k] = session.get(v);
            });

            _.map(response.namedCollections, (v, k) => {
                this.collections[k] = _.map(v, (obj) => { return session.get(obj) });
            });

            _.map(response.namedValues, (v, k) => {
                this.values[k] = v;
            });
        }
    }
}

File: Base/Workspace.Domain/Base/Client/Result.cs
namespace Allors.Workspace.Client
{
    using System.Collections.Generic;
    using System.Linq;

    using Data;

    public class Result {

        public Dictionary<string, ISessionObject> objects;
        public Dictionary<string, ISessionObject[]> collections;
        public Dictionary<string, object> values;

        public Result(ISession session, PullResponse response)
        {
            this.objects = response.namedObjects.ToDictionary(
                pair => pair.Key,
                pair => session.get(long.Parse(pair.Value)));
            this.collections = response.namedCollections.ToDictionary(
                pair => pair.Key,
                pair => pair.Value.Select(v => session.get(long.Parse(v))).ToArray());
            this.values = response.namedValues.ToDictionary(
                pair => pair.Key, 
                pair => pair.Value);
        }
    }
}